packages/codegen/src/generate.ts

```diff
@@ -756,7 +776,11 @@ export default class ApiGenerator {
 
   isTrueEnum(schema: SchemaObject, name?: string): name is string {
     return Boolean(
-      schema.enum && this.opts.useEnumType && name && schema.type !== "boolean",
+      typeof schema !== "boolean" &&
+        schema.enum &&
+        this.opts.useEnumType &&
+        name &&
+        schema.type !== "boolean",
```

Comment:
we have `schema.type !== "boolean"` two times in the conditional. can we remove this line?

Reply:
Are you sure we can do this? `typeof schema !== "boolean"` and `schema.type !== "boolean"` are two different checks.

Reply:
you're right! my bad